projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -33,41 +43,66 @@ const getMediaDetails = media => {
 		mediaWidth: mediaObject.width,
 		mediaHeight: mediaObject.height,
 		mediaSourceUrl: mediaObject.source_url,
+		fullFileSize: media.media_details.filesize,
+		mime: media.mime_type,
 	};
 };
 
 const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
 
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
-
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ validationError, setValidationError ] = useState( null );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { enabledConnections } = useSocialMediaConnections();
+
+	const { maxImageSize, getValidationError } = useMediaRestrictions( enabledConnections );
+	const allowedMediaTypes = getAllowedMediaTypes();
 
-	const mediaObject = useSelect( select =>
-		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
+	const mediaObject = useSelect(
+		select => select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } ),
+		[ attachedMedia[ 0 ] ]
 	);
+	const { mediaWidth, mediaHeight, mediaSourceUrl, fullFileSize, mime } = getMediaDetails(
+		mediaObject
+	);
+
+	useEffect( () => {
+		// Removes selected media if connection change results in invalid image
+		if ( ! mediaObject ) {
+			return;
+		}
 
-	const { mediaWidth, mediaHeight, mediaSourceUrl } = getMediaDetails( mediaObject );
+		const error = getValidationError( fullFileSize, mime );
+		if ( error ) {
+			setValidationError( error );
+			updateAttachedMedia( [] );
+		}
+	}, [ updateAttachedMedia, mediaObject, getValidationError, fullFileSize, mime ] );
 
 	const onRemoveMedia = useCallback( () => updateAttachedMedia( [] ), [ updateAttachedMedia ] );
```

Comment:
Should removing media also remove any validation errors?

Reply:
I think we cannot get to a state like that, because adding a valid media already removes the errors. So if there is an error up, that means the media couldn't get selected, or it was removed because of a connection change.

Reply:
Right, so the thinking is that there's never invalid media, there's either 1) valid media or 2) no media and a validation error. That makes sense!